js/src/name/publish.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-env mocha */
+'use strict'
+
+const each = require('async/each')
+const hat = require('hat')
+
+const { fixtures } = require('./utils')
+const { getDescribe, getIt, expect } = require('../utils/mocha')
+
+module.exports = (createCommon, options) => {
+  const describe = getDescribe(options)
+  const it = getIt(options)
+  const common = createCommon()
+
+  describe('.name.publish', function () {
+    this.timeout(50 * 1000)
+
+    const keyName = hat()
+    let ipfs
+    let nodeId
+    let keyId
+
+    before(function (done) {
+      this.timeout(60 * 1000)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist()
+
+        factory.spawnNode((err, node) => {
+          expect(err).to.not.exist()
+
+          ipfs = node
+          ipfs.id().then((res) => {
```

Comment:
Please use [`spawnNodeWithId`](https://github.com/ipfs/interface-ipfs-core/blob/34c3fc65f105528dfe17b2c9cc5ee50bea113e60/js/src/utils/spawn.js#L16) to simplify this!

Reply:
I didn't know that function! Thanks, I will use it :)